src/main/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java

```diff
@@ -73,7 +74,7 @@ private static CartDiscountDraft mapToCartDiscountDraft(
             cartDiscount.getSortOrder(),
             cartDiscount.getTarget(),
             cartDiscount.getValue())
-        .key(cartDiscount.getKey())
+        .key(StringEscapeUtils.escapeJava(cartDiscount.getKey()))
```

Comment:
Why do you escape in the draft? Draft should be created with all the special characters as provided and we want to escape on queries only like [here](https://github.com/commercetools/commercetools-sync-java/blob/c0cc6c41768bcde703616a1dbde4e2330d3df3c0/src/main/java/com/commercetools/sync/types/TypeSync.java#L109), right? Same applies for other resources.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/726/commits/cd1968be7dccb8c7d8c34f49a21866d61b80cdb5